core/src/scene/light.h

```diff
@@ -4,20 +4,21 @@
 #include "glm/glm.hpp"
 #include "util/shaderProgram.h"
 
-typedef enum {
-    DEFAULT,
-    VERTEX,
-    FRAGMENT
-} InjectionType;
-
 enum class LightType {
     UNDEFINED,
+    AMBIENT,
     DIRECTIONAL,
     POINT,
     SPOT,
     CUSTOM
 };
 
+enum class LightOrigin {
+    CAMERA,
+    GROUND,
+    WORLD
```

Comment:
`ground` is like a point dropped from the current camera location on the ground:

https://github.com/tangrams/tangram/wiki/Lights-Overview#origin

Perhaps it should be a cast ray intersecting the ground plane in this case? On web, it is (currently) always looking straight down...